### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by akhramov**
The library supports Linux, OSX and Windows, but not FreeBSD yet.

This change
* Adds FreeBSD support via `ptrace (2)` `PT_VM_ENTRY` call.
* Bumps `libc` crate version in order to get broader `ptrace (2)` call
  types support, namely `PT_VM_ENTRY`.
* Adds Cirrus-backed FreeBSD CI. `libc` crate uses Cirrus for the very
  same purpose.
* Adds mentions of FreeBSD in docstrings where appropriate.

### CODE REVIEW

**Comment by akhramov** [comment]
In order to set Cirrus CI up, please follow the guide: https://cirrus-ci.org/guide/quick-start/

I'd be happy to assist.

File: src/freebsd_maps/mod.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        let result = ptrace::read_vm_entry(pid, entry);
+
+        match result {
+            Ok(entry) => {
+                self.current = entry.pve_entry;
+                Some(entry)
+            }
+            _ => {
+                ptrace::detach(pid).ok()?;
```
Comment (benfred) [inline src/freebsd_maps/mod.rs]: Having the detach call here means that if the iterator isn't ever completely exhausted, the target process will remain locked.

Can we add to a 'Drop' trait instead?  (something like: https://github.com/benfred/py-spy/blob/1717b91d7fafc6910e1e9b76e3397a7d58e883dc/remoteprocess/src/linux/mod.rs#L170)

Comment (akhramov) [reply src/freebsd_maps/mod.rs]: Good catch, fixed!

**Review by benfred — APPROVED** [review]

**Comment by benfred** [comment]
Thanks for this! I've pushed a new version that includes these changes (0.1.6). Will get CI set up in a bit,

**Comment by akhramov** [comment]
Alright, I have updated the py-spy PR accordingly, thanks!

**Pull request merged**